### PR TITLE
completed week 3 challenges- Jeremy Duncan

### PR DESCRIPTION
**Pull request by JeremyDuncan**
# Submitting Week 3 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [✅] Coding Challenges
- [✅] Interview Questions
- [✅] React Debugging

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by JeremyDuncan** [comment]
In case you were wondering about the strange commit history, I ran into an issue where I accidentally pushed to main when I was trying to refactor. I tried to undo the changes and set main files back to initial state.

File: code-challenges.test.js [hunk 1]
```diff
@@ -2,66 +2,378 @@
...
 // Reminder: The test will call your function
 // Run the file with the following command: $ yarn jest
 
-// --------------------1) Create a function that takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence.
+//==============================================================================
+// --------------------1)👉 Create a function that takes in a number
+// (greater than 2) and returns an array that length containing the numbers
+// of the Fibonacci sequence.
+//==============================================================================
+// a)🧪 Create a test with expect statements for each of the variables provided.
```
Comment (elysemontano) [inline code-challenges.test.js:19]: Gotta say, I am liking these emoji additions!!!

File: code-challenges.test.js [hunk 2]
```diff
@@ -2,66 +2,378 @@
...
+    expect(getFibonacci(fibLength3)).toEqual(expected3);
+  });
+  it("returns that number must be greater than 2 if negative number", () => {
+    //=== Additional Edge Case ===
+    const fibLength4 = -22;
+    //=== Expected Test Results ===
+    const expected4 = "Number must be greater than 2";
 
-// b) Create the function that makes the test pass.
+    expect(getFibonacci(fibLength4)).toEqual(expected4);
```
Comment (elysemontano) [inline code-challenges.test.js:48]: Nice additions to your tests!!

File: code-challenges.test.js [hunk 3]
```diff
@@ -2,66 +2,378 @@
...
+  // loop through all object keys
+  for (const key in obj) {
+    // for each iteration push the value of the key into the array
+    numArr.push(obj[key]);
+  }
+  // return the value of the array that has been sorted from least to greatest
+  return numArr.sort(function (a, b) {
+    return a - b;
+  });
+};
```
Comment (elysemontano) [inline code-challenges.test.js:227]: For a stretch, see if you can refactor this using Object.values.

File: interview-questions.md [hunk 4]
```diff
@@ -2,54 +2,170 @@
...
 
-Then, research the question to expand on your answer. Even if you feel you have answered the question completely, there is always something more to learn. Write your researched answer in your OWN WORDS.
+Then, research the question to expand on your answer. Even if you feel you have 
+answered the question completely, there is always something more to learn. Write 
+your researched answer in your OWN WORDS.
 
 1. What is JSX? What is one syntax difference between HTML and JSX?
 
 Your answer:
+* I like to think of JSX as the marriage betwwen HTML and JavaScript. JSX is
```
Comment (elysemontano) [inline interview-questions.md:15]: As always, excellent answers in this section!!!

File: react-debugging.html [hunk 5]
```diff
@@ -25,70 +30,85 @@
...
-              return (
-                <div style={cardStyles} key={index}>
-                  {card}
-                </div>
-              )
-            })}
-          </div>
-        )
-      }
+          //👇  missing fragment*/}
```
Comment (elysemontano) [inline react-debugging.html:87]: Great work 👏

**Review by elysemontano — APPROVED** [review]
Once again, crushed this assessment!  I really like seeing all of the edge cases you added into your tests and functionality.  I left a few comments to consider, but overall, excellent work!
